### ISSUE TITLE
ES modules with primitive default export issue with static import

### ISSUE THREAD

**Issue by antfu** [unjs/jiti#342]
### Environment

Stackblitz

### Reproduction

https://stackblitz.com/edit/node-gh8a5m?file=README.md

<img width="354" alt="Screenshot 2024-11-29 at 13 51 13" src="https://github.com/user-attachments/assets/f859765d-c5dd-4357-9b19-d5bd092f9e5e">

### Describe the bug

In certain cases, when using static import for some packages with Jiti, it returns the whole module instead of the default import. In my test, I encounter that with `import isInstalledGlobally from 'is-installed-globally';`, as demonstrated in the repro.

**Comment by pi0** [comment]
babel transform:

```js
"use strict";var _isInstalledGlobally = _interopRequireDefault(await jitiImport("is-installed-globally"));function _interopRequireDefault(e) {return e && e.__esModule ? e : { default: e };}

console.log(_isInstalledGlobally.default); /* v9-ccb083195a09af69 */
```

`is-installed-globally` does not sets `__esModule` which causes it. Wondering if we should polyfill `__esModule` for native imports via a proxy.

**Comment by pi0** [comment]
should be fixed in [v2.4.1](https://github.com/unjs/jiti/releases/tag/v2.4.1)
